graylog2-web-interface/src/components/sidecars/failure-tracking/SidecarFailureTrackingList.tsx

```diff
@@ -0,0 +1,196 @@
+
+  const formatSidecarList = (sidecarRows: React.ReactNode[]) => {
+    const sidecarCollection = {
+      node_name: 'Sidecar',
+      collector: 'Collector',
+      last_seen: 'Last Seen',
+      'node_details.status.status': 'Status',
+      message: 'Error Message',
+      verbose_message: 'Verbose Message',
+    };
+
+    return (
+      <Table striped responsive>
+        <colgroup>
+          <StandardWidthCol />
+          <StandardWidthCol />
+          <StandardWidthCol />
+          <StandardWidthCol />
+          <ErrorMessageCol />
+          <VerboseMessageCol />
+        </colgroup>
+        <thead>
+          <tr>
+            {Object.keys(sidecarCollection).map((sort_key) => (
+              <th key={sort_key}>
+                {sidecarCollection[sort_key]}
+                {['node_name', 'last_seen'].includes(sort_key) && (
+                  <StyledSortIcon activeDirection={sort.field === sort_key ? sort.order : null}
+                                  onChange={() => onSortChange(sort_key)}
+                                  title={sort_key}
+                                  ascId="asc"
+                                  descId="desc" />
+                )}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>
+          {sidecarRows}
+        </tbody>
+      </Table>
+    );
+  };
+
+  const formatNoMatchingListAlert = () => {
+    const showInactiveHint = (onlyActive ? ' and/or click on "Include inactive sidecars"' : null);
+
+    return (
+      <NoSearchResult>
+        {`There are no sidecars with failures matching the search criteria. Try adjusting your search filter${showInactiveHint}.`}
+      </NoSearchResult>
+    );
+  };
+
+  const renderEmptyList = () => {
+    if (query) {
+      return formatNoMatchingListAlert();
+    }
+
+    return (
+      <NoEntitiesExist>
+        There are no sidecars with failures.
+      </NoEntitiesExist>
+    );
+  };
+
+  const sidecarRows = sidecars.map((sidecar) => <SidecarFailureTrackingRows key={sidecar.node_id} sidecar={sidecar} collectors={collectors} onShowDetails={setCollectorDetailsToShow} />);
+  const showOrHideInactive = (onlyActive ? 'Include' : 'Hide');
+  const sidecarList = (sidecarRows.length > 0 ? formatSidecarList(sidecarRows) : renderEmptyList());
+
+  return (
+    <div>
+      <div>
+        <SidecarSearchForm query={query}
+                           onSearch={onQueryChange}
+                           onReset={onQueryChange}>
+          <Button bsStyle="primary"
+                  onClick={toggleShowInactive}>
+            {showOrHideInactive} inactive sidecars
+          </Button>
+        </SidecarSearchForm>
+      </div>
+
+      <PaginatedList pageSizes={pageSizes}
+                     totalItems={pagination.total}
+                     onChange={onPageChange}>
+        <Row>
+          <Col md={12}>
+            {sidecarList}
```

Comment:
I would move the render logic in here, it is easier to read:
```suggestion
            {sidecarRows.length > 0 ? <SidecarTable rows={sidecarRows} /> : <EmptyList />}
```